go-controller/pkg/libovsdbops/db_index.go

```diff
@@ -0,0 +1,126 @@
+package libovsdbops
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+const (
+	// db ownership ExternalIDs
+	// when possible, use DbIndex and function from this file instead of directly using the following ExternalIDs
+	OwnerController = types.OvnK8sPrefix + "/controller"
+	OwnerObjectType = types.OvnK8sPrefix + "/owner-object-type"
+	OwnerNamespace  = types.OvnK8sPrefix + "/owner-namespace"
+	OwnerName       = types.OvnK8sPrefix + "/owner-name"
+	InternalID      = types.OvnK8sPrefix + "/internal-id"
+	// ExternalIDs[DbObjectID] will be used as a primary index, that is unique for every db object,
+	// and can be built based on the combination of all the other ids.
+	DbObjectID = types.OvnK8sPrefix + "/id"
```

Comment:
I am still not completely sold on generalizing the external ids in this way. I can not pinpoint why without thinking a bit more about it. Perhaps we could use a second opinion here.

But expressing the thoughts that I have right now, which come more from intuition than having an actual good reason for them other than future-proofing to avoid external-id collision in the future, I still think this should look more like:
* `<types.OvnK8sPrefix>/owner=<owner>` (there can only be one owner controller)
* `<controller>.<types.OvnK8sPrefix>/<id>=<value>` (<controller> == <owner> for the controller responsible of creating or removing the DB object, but we can have more controllers putting their own ids on a DB object)

`<owner>` can be composed of whatever we want, but correlating it to what we have here it could be composed of controller and object type, for example.

We could have as many `<id>` as we want:
* Let's decide there needs to be one for the owner controller, `id`, that needs to be a primary index.
* But let's be open ended for other `<id>`'s. Make sense since in the current proposal any of `name`, `namespace` or `internal-id` might be arbitrarily empty so why in that case would we want to have an external id like `<types.OvnK8sPrefix>/namespace=""`. Also why would we want to a generic id `internal-id` when we could be more specific than `internal-id` for that specific object type/owner ....

So, is there any drawbacks in doing it how I am thinking here?
Maybe we need to meet to properly resolve this.

Reply:
We discussed that approach in Slack, and the main drawback of using controller-specific keys in externalIDs is the need to create a separate client index for every controller+object type, in case we want to add primary or secondary index.
Therefore we decided to use some shared ids
```
"k8s.ovn.org/controller": "controller-1"
"k8s.ovn.org/object-type": "AddressSetType"
"k8s.ovn.org/name": "namespace1_name1"
"k8s.ovn.org/id": "AddressSetType.controller-1.k8s.ovn.org/namespace1_name1:ipv4"
```
`k8s.ovn.org/id` is the unique id, the composition of `k8s.ovn.org/controller` `k8s.ovn.org/object-type` and `k8s.ovn.org/name` will be used as a secondary index.

All the other ids will be prefixed with controller+object type and may only be used for predicate search, e.g.
```
"AddressSetType.controller-1.k8s.ovn.org/ipFamily": "ipv4"
```